test/server/admin/clusters_request_test.cc

```diff
@@ -0,0 +1,500 @@
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <regex>
```

Comment:
I haven't read through this in detail yet, but can you make sure we test the case where we start a /clusters request, emulate an xds update completely changing the set of clusters, and then continue the /clusters request?

Another thing it would be good to do manually is just look at the json output in a brower; Firefox has a built in json rendering mode.